src/smc-webapp/frame-editors/code-editor/code-editor-manager.ts

```diff
@@ -0,0 +1,80 @@
+/*
+Manage a collection of code editors of various files in frame trees...
+*/
+
+import { Actions, CodeEditorState } from "../code-editor/actions";
+import { get_file_editor } from "../frame-tree/register";
+import { redux } from "../../app-framework";
+
+export class CodeEditor {
+  private project_id: string;
+  private path: string;
+  private actions: Actions;
+
+  constructor(project_id: string, path: string): void {
```

Comment:
FYI, I checked travis, the `void` causes

```
 ERROR in /home/travis/build/sagemathinc/cocalc/src/smc-webapp/frame-editors/code-editor/code-editor-manager.ts(14,50)
      TS1093: Type annotation cannot appear on a constructor declaration.
```

Reply:
I wonder why we don't see that error when using typescript + webpack.

Reply:
I really don't know.